File: Tutorials/WinForms/DevExpress.MVVM/CS/Views/OrdersListView.cs
using System.Windows.Forms;
using DevExpress.Utils.MVVM.UI;
using DevExpress.XtraGrid.Views.Grid;
using WinFormsApplication.Services;
using WinFormsApplication.ViewModels;

namespace WinFormsApplication.Views {
    [ViewType("OrderListView")]
    public partial class OrdersListView : DevExpress.XtraBars.Ribbon.RibbonForm {
        public OrdersListView() {
            InitializeComponent();
            if(!mvvmContext1.IsDesignMode)
                InitializeBindings();
        }
        void InitializeBindings() {
            mvvmContext1.RegisterService(new InstantFeedbackService(ordersGridView));
            //
            var fluentApi = mvvmContext1.OfType<OrderListViewModel>();
            fluentApi.SetBinding(ordersGridControl, bs => bs.DataSource, x => x.Orders);
            fluentApi.WithEvent<RowClickEventArgs>(ordersGridView, nameof(ordersGridView.RowClick))
                .EventToCommand(x => x.Edit(),
                    (RowClickEventArgs args) => args.Clicks == 2 && args.Button == MouseButtons.Left);
            fluentApi.BindCommand(btnNew, x => x.Add());
            fluentApi.BindCommand(btnDelete, x => x.Delete());
        }
    }
}

File: Tutorials/WinForms/DevExpress.MVVM/VB/Views/OrdersListView.vb
Imports System.Windows.Forms
Imports DevExpress.Utils.MVVM.UI
Imports DevExpress.XtraGrid.Views.Grid
Imports WinFormsApplication.Services
Imports WinFormsApplication.ViewModels

Namespace WinFormsApplication.Views
	<ViewType("OrderListView")>
	Partial Public Class OrdersListView
		Inherits DevExpress.XtraBars.Ribbon.RibbonForm

		Public Sub New()
			InitializeComponent()
			If Not mvvmContext1.IsDesignMode Then
				InitializeBindings()
			End If
		End Sub
		Private Sub InitializeBindings()
			mvvmContext1.RegisterService(New InstantFeedbackService(ordersGridView))
			'
			Dim fluentApi = mvvmContext1.OfType(Of OrderListViewModel)()
			fluentApi.SetBinding(ordersGridControl, Function(bs) bs.DataSource, Function(x) x.Orders)
			fluentApi.WithEvent(Of RowClickEventArgs)(ordersGridView, NameOf(ordersGridView.RowClick)).EventToCommand(Sub(x) x.Edit(), Function(args As RowClickEventArgs) args.Clicks = 2 AndAlso args.Button = MouseButtons.Left)
			fluentApi.BindCommand(btnNew, Sub(x) x.Add())
			fluentApi.BindCommand(btnDelete, Sub(x) x.Delete())
		End Sub
	End Class
End Namespace
